### ISSUE TITLE
Remove relation(Edge) not success in StandardJanusGraphTx

### ISSUE THREAD

**Issue by jodk** [JanusGraph/janusgraph#2138]
Please post all questions and issues on [janusgraph-users](https://groups.google.com/forum/#!forum/janusgraph-users)
before opening a GitHub issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 0.5.2
- Storage Backend:hbase 
- Mixed Index Backend:
- Mailing list Thread URL:
- Steps to Reproduce: 
1. get edge (alias e1) by  traversal().V(id).outE(label) 
2. modify e1 property  
3. do something with graph 
4. remvoe e1
5. commit it

such as :
```
 @Test
    public void cc() {
        Edge next = graph.traversal().V().has("_v", 1)
                .outE("t_9")
                .next();
        next.property("_p",2);
        next.remove();
        
        graph.traversal().V().has("_v",1)
                .next().remove();
        graph.tx().commit();
    }
```

Maybe：
HashCode of edge is id : 
```
@Override
    public int hashCode() {
        return Long.hashCode(getCompareId());
    }
```
The container addedRelations in StandardJanusGraphTx:
```
  //######## Data structures that keep track of new and deleted elements
    //These data structures cannot release elements, since we would loose track of what was added or deleted
    /**
     * Keeps track of all added relations in this transaction
     */
    private final AddedRelationsContainer addedRelations;
```

when modify edge(CacheEdge) property, new edge  is added  in  addedRelations with new id , but the new edge set id to old id after add in addedRelations : 
```
private synchronized InternalRelation update() {
        StandardEdge copy = new StandardEdge(super.longId(), edgeLabel(), getVertex(0), getVertex(1), ElementLifeCycle.Loaded);
        copyProperties(copy);
        copy.remove();

        StandardEdge u = (StandardEdge) tx().addEdge(getVertex(0), getVertex(1), edgeLabel());
        if (type.getConsistencyModifier()!=ConsistencyModifier.FORK) u.setId(super.longId());
        u.setPreviousID(super.longId());
        copyProperties(u);
        setId(u.longId());
        return u;
    }
``` 
notice : hashcode is changed because of id is changed 

next step, i remove this edge ,the addedRelations.remove(relation) is trigged,but it can not be removed because of hashcode is changed
Method (StandardJanusGraphTx)as follow : 
```
public void removeRelation(InternalRelation relation) {
        Preconditions.checkArgument(!relation.isRemoved());
        relation = relation.it();
        for (int i = 0; i < relation.getLen(); i++)
            verifyWriteAccess(relation.getVertex(i));

        //Delete from Vertex
        for (int i = 0; i < relation.getLen(); i++) {
            relation.getVertex(i).removeRelation(relation);
        }
        //Update transaction data structures
        if (relation.isNew()) {
            addedRelations.remove(relation);  // remove it from ObjectHashSet
            if (TypeUtil.hasSimpleInternalVertexKeyIndex(relation)) newVertexIndexEntries.remove((JanusGraphVertexProperty) relation);
        } else {//.....}
``` 

### Stack Trace (if you have one)

```
java.lang.IllegalArgumentException: null
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:108)
	at org.janusgraph.graphdb.database.StandardJanusGraph.prepareCommit(StandardJanusGraph.java:559)
	at org.janusgraph.graphdb.database.StandardJanusGraph.commit(StandardJanusGraph.java:731)
	at org.janusgraph.graphdb.transaction.StandardJanusGraphTx.commit(StandardJanusGraphTx.java:1425)
	at org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph$GraphTransaction.doCommit(JanusGraphBlueprintsGraph.java:297)
	at org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.commit(AbstractTransaction.java:104)

```

**Comment by porunov** [comment]
Related (possibly duplicate): https://github.com/JanusGraph/janusgraph/issues/1981

**Comment by jodk** [comment]
> Related (possibly duplicate): #1981

duplicate #1981  ， the root cause is reset id of InternalRelation when it is saved to ObjectHashSet

**Comment by li-boxuan** [comment]
This is not a duplicate of #1981. I reproduced this on v0.5 branch, but not v0.4 branch  (the #1981 bug can be reproduced in both v0.5 and v0.4):

```java
    @Test
    public void testEdgeRemoval() {
        Vertex v = graph.traversal().addV().property("_v", 1).next();
        Vertex v2 = graph.traversal().addV().property("_v", 2).next();
        v.addEdge("t_9", v2, "_p", 1);
        graph.tx().commit();

        Edge next = graph.traversal().V().has("_v", 1)
            .outE("t_9")
            .next();
        next.property("_p",2);
        next.remove();

        graph.traversal().V().has("_v",1)
            .next().remove();
        graph.tx().commit();
    }
```

Probably related to the change in https://github.com/JanusGraph/janusgraph/pull/1701, which was released in 0.5.0
